test/functorch/test_control_flow.py

```diff
@@ -762,5 +764,256 @@ def foo(x):
+
+    def test_capture_global_num(self):
+        mock = MockBackend()
+        x = torch.zeros([])
+
+        @torch.compile(backend=mock)
+        def f(x):
+            return wrap(lambda x: x + global_num, x)
+
+        global global_num
+        result = f(x)
+        self.assertEqual(result, x + global_num)
+        self.assertEqual(len(mock.graphs), 1)
+        gm = mock.graphs[0]
+        # Numbers don't get lifted
+        self.assertIsNotNone(re.search(r'wrap\(\w+, \w+\);', gm.code))
+
+        # Check that we still guard on the number
+        global_num = torch.randn([]).item()
+        result = f(x)
+        self.assertEqual(result, x + global_num)
+
+    def test_capture_input_num(self):
+        mock = MockBackend()
+        x = torch.zeros([])
+        y = 3.14
+
+        @torch.compile(backend=mock)
+        def f(x, y):
+            return wrap(lambda x: x + y, x)
+
+        result = f(x, y)
+        self.assertEqual(result, x + y)
+        self.assertEqual(len(mock.graphs), 1)
+        gm = mock.graphs[0]
+        # Numbers don't get lifted
+        self.assertIsNotNone(re.search(r'wrap\(\w+, \w+\);', gm.code))
+
+    def test_fallback_on_graph_break_simple(self):
+        # In the future, there should be a per-HigherOrderOperator switch
+        # on whether or not to fallback or raise a loud error.
+        # For now we just fallback by default.
+        mock = MockBackend()
+        x = torch.randn([])
+
+        def inner(x):
+            y = x.sin()
+            torch._dynamo.graph_break()
+            z = y.sin()
+            return z
+
+        @torch.compile(backend=mock)
+        def f(x):
+            return wrap(inner, x)
+
+        result = f(x)
+        self.assertEqual(result, inner(x))
+        # It's unclear if this is correct: dynamo graph breaks on wrap but
+        # then interposes on wrap.__call__, which invokes fn(*args),
+        # leading to two graphs being compiled
+        self.assertEqual(len(mock.graphs), 2)
+
+    def test_fallback_on_graph_break_complicated(self):
+        mock = MockBackend()
+        x = torch.randn([])
+
+        def inner(x):
+            y = x.sin()
+            y = y * global_var
+            torch._dynamo.graph_break()
+            z = y.sin()
+            return z
+
+        @torch.compile(backend=mock)
+        def f(x):
+            x = x.clone()
+            result = wrap(inner, x)
+            return result.clone()
+
+        result = f(x)
+        self.assertEqual(result, inner(x))
+        # It's unclear if this is correct: dynamo graph breaks on wrap but
+        # then interposes on wrap.__call__, which invokes fn(*args),
+        # leading to four graphs being compiled: clone, sin, sin, clone
+        self.assertEqual(len(mock.graphs), 4)
+
+    # TODO(rzou): investigate failure
+    @unittest.expectedFailure
+    def test_fallback_on_modules(self):
+        # We can likely support this in the future, I just don't want to deal
+        # with it right now
```

Comment:
Claim: nn.Modules should not appear in the body graph of HigherOrderOperators because it is implicit state that will make it difficult for future transformation (e.g. lowering control flow ops to inductor, vmap). 

In cond/map today, they do appear, so we need to fix that (see example below for map).
```py
class MockBackend:
    def __init__(self):
        self.graphs = []

    def __call__(self, gm: torch.fx.GraphModule, example_inputs):
        self.graphs.append(gm)
        return gm.forward

mock = MockBackend()
mod = torch.nn.Linear(3, 3)
from functorch.experimental.control_flow import map
x = torch.randn(2, 3, 3)

@torch.compile(backend=mock)
def f(x):
    return map(lambda x: mod(x), x)

result = f(x)
```
produces the following graph:
```
(Pdb) print(mock.graphs)
[GraphModule(
  (cond_body_0): GraphModule(
    (mod): Linear(in_features=3, out_features=3, bias=True)
  )
)]
```